Design/new-mangling-approach.md

```diff
@@ -0,0 +1,74 @@
+## Name Mangling
+
+### 1. Registration of Types
+
+- Types are registered to track unique identifiers for variables.
+
+### 2. Registration of Names
+
+- Names (e.g., variable basenames) are registered within scopes.
+- Each name is associated with a type, allowing for type-based disambiguation during resolution.
+
+### 3. Assigning Output Versions to Types
+
+- For each type, select a single output version from $\{short, medium, long\}$ used for all variables of that type.
```

Comment:
Why is this desireable?  It seems like the key slicing should be by name: for a name `x` we may decide on different strategies for disambiguating it (by type, by scope, by argument (Kotlin) types).  Why make this choice per type?